.github/workflows/lab-validation-pr.yml

```diff
@@ -0,0 +1,172 @@
+name: Lab Validation for PR
+
+on:
+  # Trigger when label is added to PR or new commits are pushed (if label exists)
+  pull_request:
+    types: [labeled, synchronize]
+
+jobs:
+  # Check if we should run based on trigger type
+  check-trigger:
+    runs-on: ubuntu-latest
+    outputs:
+      should_run: ${{ steps.check.outputs.should_run }}
+      pr_number: ${{ steps.check.outputs.pr_number }}
+    steps:
+    - name: Check trigger conditions
+      id: check
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        SHOULD_RUN="false"
+        PR_NUMBER=""
+        
+        if [ "${{ github.event.action }}" = "labeled" ]; then
+          # Check if 'lab-validation' label was added
+          if [ "${{ github.event.label.name }}" = "lab-validation" ]; then
+            SHOULD_RUN="true"
+            PR_NUMBER="${{ github.event.number }}"
+            echo "🏷️ Label 'lab-validation' added to PR #$PR_NUMBER"
+          fi
+        elif [ "${{ github.event.action }}" = "synchronize" ]; then
+          # Check if PR has 'lab-validation' label when new commits are pushed
+          LABELS=$(gh api repos/batfish/batfish/pulls/${{ github.event.number }}/labels --jq '.[].name')
```

Comment:
Is there really no other way to get labels from inside github actions?!

Reply:
Unfortunately, GitHub Actions doesn't provide PR labels directly in the event data for 'synchronize' events. The event only contains commit information, not the current labels. The GitHub API call is the standard approach - you'll see this pattern in many popular actions like 'actions/labeler' and similar workflows that need to check current PR state.